diesel/src/query_builder/select_by_statement/dsl_impls.rs

```diff
@@ -0,0 +1,365 @@
+// use super::BoxedSelectByStatement;
+use crate::associations::HasTable;
+use crate::deserialize::TableQueryable;
+// use crate::backend::Backend;
+// use crate::expression::nullable::Nullable;
+use crate::expression::*;
+use crate::insertable::Insertable;
+use crate::query_builder::insert_statement::InsertFromSelect;
+use crate::query_builder::{Query, SelectByStatement, SelectQuery};
+// use crate::query_dsl::boxed_dsl::BoxedDsl;
+use crate::query_dsl::methods::*;
+use crate::query_dsl::*;
+// use crate::query_source::joins::{Join, JoinOn, JoinTo};
+use crate::query_source::joins::JoinTo;
+// use crate::query_source::QuerySource;
+use crate::sql_types::Bool;
+
+impl<S, STMT, Rhs, Kind, On> InternalJoinDsl<Rhs, Kind, On> for SelectByStatement<S, STMT>
```

Comment:
I think `Stmt` instead of `STMT` would be a more idiomatic rust style. (Applies for all similar cases in this file)